lib/ReactViews/Analytics/RectangleParameterEditor.tsx

```diff
@@ -1,41 +1,61 @@
-"use strict";
-
-import React from "react";
-
-import PropTypes from "prop-types";
-
+import React, { type ChangeEvent } from "react";
 import defined from "terriajs-cesium/Source/Core/defined";
-
 import Styles from "./parameter-editors.scss";
-
 import CesiumMath from "terriajs-cesium/Source/Core/Math";
 import Cartographic from "terriajs-cesium/Source/Core/Cartographic";
 import Ellipsoid from "terriajs-cesium/Source/Core/Ellipsoid";
-import UserDrawing from "../../Models/UserDrawing";
-import { withTranslation } from "react-i18next";
+import {
+  type TFunction,
+  WithTranslation,
+  withTranslation
+} from "react-i18next";
 import { observer } from "mobx-react";
-import { runInAction } from "mobx";
+import { makeObservable, runInAction } from "mobx";
+import { BaseModel } from "../../Models/Definition/Model";
+import RectangleParameter from "../../Models/FunctionParameters/RectangleParameter";
 import CommonStrata from "../../Models/Definition/CommonStrata";
+import Terria from "../../Models/Terria";
+import UserDrawing from "../../Models/UserDrawing";
+import ViewState from "../../ReactViewModels/ViewState";
+
+interface RectangleParameterEditorProps extends WithTranslation {
```

Comment:
I mimicked some other component with the `extends WithTranslation`, but I have no idea if that is how it should be done. I see #7398 removes a couple of `extends WithT` and just puts in `t` instead, so maybe this isn't required?